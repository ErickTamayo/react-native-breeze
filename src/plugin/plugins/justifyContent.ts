import { PluginFunction } from "./types";

export type PluginGroups = {
  justify: "start" | "end" | "center" | "between" | "around" | "evenly";
};

export const pattern = /^justify-(?<justify>start|end|center|between|around|evenly)$/;

export const plugin: PluginFunction<PluginGroups> = ({ groups }) => {
  const { justify } = groups;

  switch (justify) {
    case "start":
      return { justifyContent: "flex-start" };
    case "end":
      return { justifyContent: "flex-end" };
    case "center":
      return { justifyContent: "center" };
    case "between":
      return { justifyContent: "space-between" };
    case "around":
      return { justifyContent: "space-around" };
    case "evenly":
      return { justifyContent: "space-evenly" };
  }
};
