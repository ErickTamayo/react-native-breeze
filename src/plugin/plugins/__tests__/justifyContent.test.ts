import { pattern, plugin } from "../justifyContent";
import {
  shouldEvaluateTheCorrectPatternTest,
  generateInput,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";

describe("justifyContent", () => {
  // prettier-ignore
  const shouldMatch = generateInput('justify', ['start', 'end', 'center', 'between', 'around', 'evenly'])

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["unknown", "justify-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
