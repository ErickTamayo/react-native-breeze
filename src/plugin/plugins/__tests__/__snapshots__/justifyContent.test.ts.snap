// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`justifyContent Should parse (justify-around) correctly 1`] = `
Object {
  "justifyContent": "space-around",
}
`;

exports[`justifyContent Should parse (justify-between) correctly 1`] = `
Object {
  "justifyContent": "space-between",
}
`;

exports[`justifyContent Should parse (justify-center) correctly 1`] = `
Object {
  "justifyContent": "center",
}
`;

exports[`justifyContent Should parse (justify-end) correctly 1`] = `
Object {
  "justifyContent": "flex-end",
}
`;

exports[`justifyContent Should parse (justify-evenly) correctly 1`] = `
Object {
  "justifyContent": "space-evenly",
}
`;

exports[`justifyContent Should parse (justify-start) correctly 1`] = `
Object {
  "justifyContent": "flex-start",
}
`;
